pkg/registry/generic/etcd/etcd.go

```diff
@@ -398,6 +398,11 @@ func (e *Etcd) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 	return e.finalizeDelete(out, true)
 }
 
+// TODO Should this be here, or should each resource have it's own DeleteCollection()?
+func (e *Etcd) DeleteCollection(ctx api.Context) (runtime.Object, error) {
```

Comment:
I think the behavior here is resource specific, so maybe let me pass a function to delegate the implementation of the desired behavior similar to what is done for the KeyFuncs, I suspect we will have a couple common functions.

For example, I see no issue for Events, Services, ReplicationControllers, ResourceQuotas, LimitRanges all supporting the ability to delete the entire collection even if it contains items - especially during namespace clean-up.  The function here would just do a etcd delete on KeyRootFunc, and things would just cascade nicely.

For Namespaces, I do not want to support DeleteCollection at this time, so I would want an implementation that results in the equivalent of a Forbidden response.

For Pods, I think we want to preserve the ability to do maintain support for graceful deletion.  As a result, I think pods would want a version of this function that would fail if the collection had any child nodes.  In practice, this means the namespace_controller would have to delete each pod in turn, and then once all pods are removed, then it would call DeleteCollection (since its empty now) to delete the root folder for pods in that namespace.  I would be interested in @smarterclayton take on this to see if I am being overly concerned.

For Nodes, I am not sure we want to support the operation at all.